services/streammanager/googlecloudfunction/googlecloudfunction.go

```diff
@@ -0,0 +1,261 @@
+	if err != nil {
+		fmt.Println("Error creating request:", err)
+		return
+	}
+
+	// Set the appropriate headers
+	req.Header.Set("Content-Type", "application/json")
+	if requireAuthentication {
+		ts, err := idtoken.NewTokenSource(ctx, functionUrl, option.WithCredentialsJSON([]byte(credentials)))
+		if err != nil {
+			fmt.Print("failed to create NewTokenSource: %w", err)
+		}
+
+		// Get the ID token, to make an authenticated call to the target audience.
+		token, err := ts.Token()
+		if err != nil {
+			fmt.Print("failed to receive token: %w", err)
+		}
+		req.Header.Add("Authorization", "Bearer "+token.AccessToken)
+	}
+
+	// Make the request using the client
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		responseMessage = err.Error()
+		statusCode = 400
+		if errors.Is(err, context.DeadlineExceeded) {
+			statusCode = 504
+		}
+		respStatus = "Failure"
+		responseMessage = "[GOOGLE_CLOUD_FUNCTION] error :: Function call was not executed (Not Modified :: " + responseMessage
+		pkgLogger.Errorf("[GOOGLE_CLOUD_FUNCTION] error while calling the Gen2 function :: %v", err)
+		return statusCode, respStatus, responseMessage
+	}
+	defer resp.Body.Close()
+
+	if resp.Status == "OK" {
+		respStatus = "Success"
+		responseMessage = "[GoogleCloudFunction] :: Message Payload inserted with messageId :: " + parsedJSON.Get("id").String()
+	}
+	fmt.Print(resp.Status)
+	return 200, respStatus, responseMessage
+}
+
+// Initialize the Cloud Functions API client using service account credentials.
+func generateService(opts ...option.ClientOption) (*cloudfunctions.Service, error) {
+	ctx := context.Background()
+
+	cloudFunctionService, err := cloudfunctions.NewService(ctx, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("[GoogleCloudFunction] error  :: Unable to create cloudfunction service :: %w", err)
+	}
+	return cloudFunctionService, err
+}
+
+func getFunctionName(url string) (functionName string) {
+	// Define a regular expression pattern to match URLs between "https://" and dot (.)
+	pattern := `https://(.*?)\.`
+
+	// Compile the regular expression pattern
+	regExp := regexp.MustCompile(pattern)
+
+	// Find all matches in the input string
+	matches := regExp.FindAllStringSubmatch(url, -1)
+
+	// Extract the captured groups
+	var resultArray []string
+	for _, match := range matches {
+		if len(match) > 1 {
+			resultArray = append(resultArray, match[1])
+		}
+	}
+
+	// Combine the first two elements
+	splitValues := strings.Split(resultArray[0], "-")
+	// Join the first two values with "-" and assign to REGION
+	REGION := strings.Join(splitValues[:2], "-")
+
+	// Join the rest of the values with "-" and assign to PROJECT_ID
+	PROJECT_ID := strings.Join(splitValues[2:], "-")
+
+	var FUNCTION_NAME string
+
+	index := strings.Index(url, "cloudfunctions.net/")
+
+	if index != -1 {
+		// Extract the substring after "cloudfunctions.net/"
+		FUNCTION_NAME = url[index+len("cloudfunctions.net/"):]
+	} else {
+		fmt.Println("No match found")
```

Comment:
Can we use a logger here instead ?

Reply:
Alternate way:
1. remove the else block
2. Above `if index != -1 {`, can we add a logger which outputs the index value at debug scope `pkglogger.Debugf("SearchIndex: %d", index)`